test_conformance/subgroups/subgroup_common_templates.h

```diff
@@ -388,11 +390,57 @@ template <typename Ty> bool is_floating_point()
         || std::is_same<Ty, subgroups::cl_half>::value;
 }
 
+// limit possible input values to avoid arithmetic rounding/overflow issues.
+// for each subgroup values defined different values
+// for rest of workitems set 1
+// shuffle values
+template <typename Ty>
+void fill_and_shuffle_safe_values(std::vector<cl_ulong> &safe_values,
+                                  const int &sb_size)
+{
+    // below values are 0.5f, 0.25f, 2.0f, 4.0f
+    std::vector<cl_ulong> safe_half_values{ 0x3800, 0x3400, 0x4000, 0x4400 };
+    // max possible value is == 5040
+    std::vector<cl_ulong> safe_other_types_values{ 2, 3, 4, 5, 6, 7 };
+
+    cl_ulong set_one = 0x3c00;
```

Comment:
Minor: I'd find this easier to follow if the intial value was `1` and was only overridden to `0x3c00` for halfs (though see comment above).

Reply:
fixed